pinot-common/src/main/java/org/apache/pinot/common/request/BrokerRequest.java

```diff
@@ -2502,30 +2481,30 @@ public void read(org.apache.thrift.protocol.TProtocol prot, BrokerRequest struct
       }
       if (incoming.get(12)) {
         {
-          org.apache.thrift.protocol.TMap _map119 = new org.apache.thrift.protocol.TMap(org.apache.thrift.protocol.TType.STRING, org.apache.thrift.protocol.TType.STRING, iprot.readI32());
-          struct.debugOptions = new java.util.HashMap<java.lang.String,java.lang.String>(2*_map119.size);
-          @org.apache.thrift.annotation.Nullable java.lang.String _key120;
-          @org.apache.thrift.annotation.Nullable java.lang.String _val121;
-          for (int _i122 = 0; _i122 < _map119.size; ++_i122)
+          org.apache.thrift.protocol.TMap _map127 = new org.apache.thrift.protocol.TMap(org.apache.thrift.protocol.TType.STRING, org.apache.thrift.protocol.TType.STRING, iprot.readI32());
+          struct.debugOptions = new java.util.HashMap<java.lang.String,java.lang.String>(2*_map127.size);
+          @org.apache.thrift.annotation.Nullable java.lang.String _key128;
+          @org.apache.thrift.annotation.Nullable java.lang.String _val129;
+          for (int _i130 = 0; _i130 < _map127.size; ++_i130)
           {
-            _key120 = iprot.readString();
-            _val121 = iprot.readString();
-            struct.debugOptions.put(_key120, _val121);
+            _key128 = iprot.readString();
+            _val129 = iprot.readString();
+            struct.debugOptions.put(_key128, _val129);
           }
         }
         struct.setDebugOptionsIsSet(true);
       }
       if (incoming.get(13)) {
         {
-          org.apache.thrift.protocol.TMap _map123 = new org.apache.thrift.protocol.TMap(org.apache.thrift.protocol.TType.STRING, org.apache.thrift.protocol.TType.STRING, iprot.readI32());
-          struct.queryOptions = new java.util.HashMap<java.lang.String,java.lang.String>(2*_map123.size);
-          @org.apache.thrift.annotation.Nullable java.lang.String _key124;
-          @org.apache.thrift.annotation.Nullable java.lang.String _val125;
-          for (int _i126 = 0; _i126 < _map123.size; ++_i126)
+          org.apache.thrift.protocol.TMap _map131 = new org.apache.thrift.protocol.TMap(org.apache.thrift.protocol.TType.STRING, org.apache.thrift.protocol.TType.STRING, iprot.readI32());
+          struct.queryOptions = new java.util.HashMap<java.lang.String,java.lang.String>(2*_map131.size);
+          @org.apache.thrift.annotation.Nullable java.lang.String _key132;
+          @org.apache.thrift.annotation.Nullable java.lang.String _val133;
+          for (int _i134 = 0; _i134 < _map131.size; ++_i134)
           {
-            _key124 = iprot.readString();
-            _val125 = iprot.readString();
-            struct.queryOptions.put(_key124, _val125);
+            _key132 = iprot.readString();
+            _val133 = iprot.readString();
+            struct.queryOptions.put(_key132, _val133);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 2498:2509, 2483:2494.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.